File: Angular.Microservice/Angular.Microservice/Controllers/SupplierController.cs
using Angular.Microservice.BussinessLogic.Interface;
using Angular.Microservice.DataAccess.Context;
using Angular.Microservice.DataAccess.Model;
using Angular.Microservice.DataAccess.Param;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Web;
using System.Web.Mvc;

namespace Angular.Microservice.Controllers
{
    public class SupplierController : Controller
    {
        private MyContext myContext = null;
        Supplier param = new Supplier();

        private readonly ISupplierService _supplierService;
        public SupplierController(ISupplierService supplierService)
        {
            _supplierService = supplierService;
        }
        public SupplierController()
        {
            myContext = new MyContext();
        }
        // GET: Supplier
        public JsonResult GetSuppliers()
        {
            List<Supplier> listSuppliers = myContext.Suppliers.ToList();
            //var listSuppliers = _supplierService.Get();
            return Json(new { list = listSuppliers }, JsonRequestBehavior.AllowGet);
            
        }
        public JsonResult GetSupplierById(int id)
        {
            Supplier supplier = myContext.Suppliers.Where(x => x.Id == id).SingleOrDefault();
            //var supplier = _supplierService.Get(id);
            return Json(new { supplier = supplier }, JsonRequestBehavior.AllowGet);
        }

        public JsonResult AddSupplier(Supplier supplier)
        {
            if(supplier.Name != null)
            {
                param.Name = supplier.Name;
                myContext.Suppliers.Add(param);
                myContext.SaveChanges();
                return Json(new { status = "Supplier Added Successfully" });
            }
            return Json(new { status = "Adding Supplier Failed" });
        }

        public JsonResult UpdateSupplier(Supplier supplier)
        {
            param.Name = supplier.Name;
            myContext.Entry(param).State = System.Data.Entity.EntityState.Modified;
            myContext.SaveChanges();
            return Json(new { status = "Supplier Updated Successfully" });
        }

        public JsonResult DeleteSupplier(int id)
        {
            param.IsDelete = true;
            myContext.Entry(param).State = System.Data.Entity.EntityState.Modified;
            myContext.SaveChanges();
            return Json(new { status = "Supplier Was Soft Deleted Successfully" });
        }
    }

}

File: Angular.Microservice/Angular.Microservice/Scripts/SupplierController/SupplierController.js
angular.module("SupplierController", []).
controller("MainController", function ($scope, SupplierService) {
    $scope.message = "Main Controller";

    SupplierService.GetSupplierFromDB().then(function (d) {

        $scope.listSuppliers = d.data.list;
    })

    $scope.DeleteSupplier = function (id, index) {
        $scope.listSuppliers.splice(index, 1);
        SupplierService.DeleteSupplier(id);
    }
}).
controller("AddSupplierController", function ($scope, SupplierService) {
    $scope.message = "Add Supplier Details";

    $scope.AddSupplier = function () {
        SupplierService.AddSupplier($scope.supplierParam);
    }
}).
controller("EditSuppllierController", function ($scope, SupplierService, $routeParams) {
    $scope.message = "Update Supplier Details";

    var id = $routeParams.id;

    SupplierService.GetSupplierByID(id).then(function (d) {
        $scope.supplier = d.data.supplierParam;
    })

    $scope.UpdateSupplier = function () {
        SupplierService.UpdateSupplier($scope.supplierParam);
    }
}).
factory("SupplierService", ["$http", function ($http) {

    var fac = {};

    fac.GetSupplierFromDB = function () {
        return $http.get("/Supplier/GetSuppliers");
    }

    fac.GetSupplierByID = function (id) {
        return $http.get("/Supplier/GetSupplierById", { params: { id: id } });
    }

    fac.AddSupplier = function (supplier) {
        $http.post("/Supplier/AddSupplier", supplier).success(function (response) {
            alert(response.status);
        })
    }

    fac.UpdateSupplier = function (supplier) {
        $http.post("/Supplier/UpdateSupplier", supplier).success(function (response) {
            alert(response.status);
        })
    }

    fac.DeleteSupplier = function (id) {
        $http.post("/Supplier/DeleteSupplier", { id: id }).success(function (response) {
            alert(response.status);
        })
    }

    return fac;
}])